db/migrate_foreman/20120824142048_add_some_indexes.foreman_puppet.rb

```diff
@@ -13,9 +13,9 @@ def up
   end
 
   def down
-    remove_foreign_key 'host_classes', 'hosts', name: 'host_classes_host_id_fk'
-    remove_foreign_key 'host_classes', 'puppetclasses', name: 'host_classes_puppetclass_id_fk'
-    remove_foreign_key 'operatingsystems_puppetclasses', 'operatingsystems', name: 'operatingsystems_puppetclasses_operatingsystem_id_fk'
-    remove_foreign_key 'operatingsystems_puppetclasses', 'puppetclasses', name: 'operatingsystems_puppetclasses_puppetclass_id_fk'
+    remove_foreign_key('host_classes', 'hosts') if foreign_key_exists?(:host_classes, :hosts)
```

Comment:
Any idea whether this error can occur for the other migrations?

Reply:
no idea if it can, it didn't in Satellite usecase, but I can add the condition to all of them :)

Reply:
Won't hurt I guess :) Any idea why the error occurred in the first place?

Reply:
I believe it was because of the FK was created with the table alrady and thus had a default name, not the name given by us.